### PR TITLE
RELATED: ONE-5027 Add responsive legend examples

### PR DESCRIPTION
**Pull request by pbenes**
- Add examples for responsive legend with pre-defined and custom
  (draggable) size
- Unify bar chart and headline chart responsive examples with responsive
  legend examples.

JIRA: ONE-5027

---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3697/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5316/ : FAILURE in 5m 54s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3700/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5320/ : SUCCESS in 12m 19s

File: examples/sdk-examples/src/examples/chartResponsiveness/index.tsx [hunk 1]
```diff
@@ -5,19 +5,28 @@ import React from "react";
...
 
 import HeadlineResponsiveExampleSrcJS from "!raw-loader!../../../examplesJS/chartResponsiveness/HeadlineResponsiveExample";
 import BarChartResponsiveExampleSrcJS from "!raw-loader!../../../examplesJS/chartResponsiveness/BarChartResponsiveExample";
+import ColumnChartResponsiveLegendExampleSrcJS from "!raw-loader!../../../examplesJS/chartResponsiveness/ColumnChartResponsiveLegendExample";
 
 export const ChartResponsiveness: React.FC = () => (
     <div>
         <h1>Chart Responsiveness</h1>
-        <p>These examples show improved responsive behaviour of charts.</p>
+        <p>These examples show improved responsive behaviour of charts and headline.</p>
```
Comment (evgeniabogranskaya) [inline examples/sdk-examples/src/examples/chartResponsiveness/index.tsx:21]: These examples show responsive behavior of charts.

Comment (evgeniabogranskaya) [reply examples/sdk-examples/src/examples/chartResponsiveness/index.tsx:21]: I suggest that we do not mention "improved" - compared to what? Also, let's not specify headlines as a separate category. Having it as a separate example makes it stand out sufficiently.

File: examples/sdk-examples/src/examples/chartResponsiveness/index.tsx [hunk 2]
```diff
@@ -5,19 +5,28 @@ import React from "react";
...
 
         <h2 id="headline">Headline</h2>
+        <p>
+            Headline shows pagination for smallest size. If there is enough vertical space, it displays
+            additional values below each other. When there is enough horizontal space, it uses the default
+            side-by-side layout.
```
Comment (evgeniabogranskaya) [inline examples/sdk-examples/src/examples/chartResponsiveness/index.tsx:27]: When rendered in the smallest possible size, headlines enable pagination. If there is enough vertical space, a headline displays additional values one under the other. If there is enough horizontal space, it uses the default side-by-side layout.

File: examples/sdk-examples/src/examples/chartResponsiveness/index.tsx [hunk 3]
```diff
@@ -28,12 +37,25 @@ export const ChartResponsiveness: React.FC = () => (
         <h2 id="bar-chart">Visual Components</h2>
         <p>
             Small visual components will have parts of the chart hiden to improve user responsiveness
-            experience.
+            experience. Various widget heights can be seen in this example. Smallest has only the axis title
+            visible, middle has only the axis labels visible and the largest has both axis title and labels
+            visible.
```
Comment (evgeniabogranskaya) [inline examples/sdk-examples/src/examples/chartResponsiveness/index.tsx:42]: When rendered in a smaller size, visual components hide different sections to improve a visual user experience. You can see how the chart changes when you switch between different heights. With the smallest value, the chart has only the axis title visible. With the middle value, the chart has only the axis labels visible. With the largest value, the chart has both the axis title and labels visible.

File: examples/sdk-examples/src/examples/chartResponsiveness/index.tsx [hunk 4]
```diff
@@ -28,12 +37,25 @@ export const ChartResponsiveness: React.FC = () => (
...
         <ExampleWithSource
             for={BarChartResponsiveExample}
             source={BarChartResponsiveExampleSrc}
             sourceJS={BarChartResponsiveExampleSrcJS}
         />
+        <h2>Visual Components: responsive legend</h2>
+        <p>
+            Demonstration of legend responsive behaviour. Try changing legend responsive config for different
+            chart dimensions. When configured, it can be seen that the legend has forced position and pop-up
+            functionality for smaller dimensions.
```
Comment (evgeniabogranskaya) [inline examples/sdk-examples/src/examples/chartResponsiveness/index.tsx:53]: <h2>Responsive Legend</h2>
<p>To see a responsive legend, try various combinations of the legend position/responsiveness and the container dimensions.
When the responsiveness is set to "autoPositionWithPopup", you can see that the legend has a forced position and appears as a popup in smaller containers.

**Comment by evgeniabogranskaya** [comment]
LGTM

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3715/ : SUCCESS in 9s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5337/ : FAILURE in 5m 45s

**Review by kandl — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3719/ : SUCCESS in 11s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5341/ : SUCCESS in 12m 35s

**Comment by yenkins** [comment]
### Sonar scan result
 - No issue found. Congratulations!

More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR1368

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1089/ : SUCCESS in 1m 10s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1359/ : SUCCESS in 17s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/85/ : SUCCESS in 6m 46s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5342/ : SUCCESS in 12m 31s

**Pull request merged**